sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -256,6 +256,21 @@ class FindDataSourceTable(sparkSession: SparkSession) extends Rule[LogicalPlan]
   }
 }
 
+case class DetermineDataSourceTableStats(session: SparkSession) extends Rule[LogicalPlan] {
+
+  override def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
+    case logicalRelation @ LogicalRelation(_, _, catalogTable, _) if catalogTable.nonEmpty &&
+        catalogTable.forall(DDLUtils.isDatasourceTable) && catalogTable.forall(_.stats.isEmpty) =>
+      val sizeInBytes = if (session.sessionState.conf.fallBackToHdfsForStatsEnabled) {
```

Comment:
Looks like this code will be a duplicate at https://github.com/apache/spark/pull/24712. Are there more places that similar patterns happen? Let's do that PR first and deduplicate it if there are some more places to deduplicate.

Reply:
I'm move `DetermineTableStats` from `HiveStrategies` to `DataSourceStrategy` to reduce duplicate.